src/reduce_dims.cpp

```diff
@@ -37,17 +35,33 @@ bool reduce_dim(std::vector<shape>& shapes, std::size_t n)
     return true;
 }
 
+void reduce_dim1(std::vector<shape>& shapes)
```

Comment:
This looks similar to `squeeze` op on last axis.

Reply:
Yea, since the other `reduce_dim` doesn't work on the last dimension.